receiver/otlpreceiver/otlp_test.go

```diff
@@ -542,6 +542,31 @@ func TestHTTPNewPortAlreadyUsed(t *testing.T) {
 	require.Error(t, r.Start(context.Background(), componenttest.NewNopHost()))
 }
 
+func TestReceiverLifecycle(t *testing.T) {
+	endpointGrpc := testutil.GetAvailableLocalAddress(t)
+	endpointHTTP := testutil.GetAvailableLocalAddress(t)
+
+	tracesSink := new(consumertest.TracesSink)
+	metricsSink := new(consumertest.MetricsSink)
+
+	// Create OTLP receiver with gRPC and HTTP protocols.
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig().(*Config)
+	cfg.GRPC.NetAddr.Endpoint = endpointGrpc
+	cfg.HTTP.Endpoint = endpointHTTP
+
+	fstReceiver := newReceiver(t, factory, cfg, tracesSink, metricsSink)
+	mh := newAssertNoErrorHost(t)
+	require.NoError(t, fstReceiver.Start(context.Background(), mh))
+
+	sndReceiver := newReceiver(t, factory, cfg, tracesSink, metricsSink)
```

Comment:
This is not safe, because if you start this receiver before shutting down the old one you will override the next consumer pointer which is not safe.

Reply:
Changed lifecycle per #2757 - so no more overlapping registration of consumers.